### PR TITLE
chore: expose iam actions map without affecting public API

### PR DESCRIPTION
**Pull request by awsluja**
## Problem
We need to maintain a mapping of actions, meta actions, and their iam actions in the docs. If the mapping is exported we can use it in the docs site to render a table.

The goal of this change is to do so without having to modify the public API.

It can be imported by adding this package, @aws-amplify/backend-auth, to your package.json & importing it like:

`import { iamActionMap } from '@aws-amplify/backend-auth/internal/iam_actions';
`

**Issue number, if available:**

## Changes

**Corresponding docs PR, if applicable:**

## Checklist

- [ ] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [ ] If this PR requires a docs update, I have linked to that docs PR above.
- [ ] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by Amplifiyer — APPROVED** [review]

File: packages/backend-auth/package.json [hunk 1]
```diff
@@ -10,6 +10,11 @@
       "types": "./lib/index.d.ts",
       "import": "./lib/index.js",
       "require": "./lib/index.js"
+    },
+    "./internal/iam_actions": {
+      "types": "./lib/internal/iam_actions.d.ts",
+      "import": "./lib/internal/iam_actions.js",
+      "require": "./lib/internal/iam_actions.js"
```
Comment (sobolk) [inline packages/backend-auth/package.json:17]: will this show up in IDE syntax suggestions?

Comment (awsluja) [reply packages/backend-auth/package.json:17]: You mean like this? or from the amplify DX? 
image

Comment (sobolk) [reply packages/backend-auth/package.json:17]: > or from the amplify DX?

From amplify DX. Are customers going to see possibility to import these from `@aws-amplify/backend-auth/internal` ?

Comment (awsluja) [reply packages/backend-auth/package.json:17]: Ok, so if the question is about doing this:
image

There is no intellisense and you must manually specify the /internal/iam_actions path to be able to import it

Comment (awsluja) [reply packages/backend-auth/package.json:17]: https://github.com/aws-amplify/amplify-backend/assets/110861985/9ca997df-d782-4170-8ba7-4b96fe95045e

Comment (sobolk) [reply packages/backend-auth/package.json:17]: I get this 
image


When trying to use similar export from here https://github.com/aws-amplify/amplify-backend/blob/4dd9d5a35c378895d1360c15a3b7ad1f09cc7653/packages/client-config/package.json#L14-L18

Comment (edwardfoyle) [reply packages/backend-auth/package.json:17]: I'm okay with this as-is, but if we want an even stronger suggestion that customers shouldn't use this, we could namespace it `internal-only-do-not-use/iam-actions` or something to that effect

Comment (awsluja) [reply packages/backend-auth/package.json:17]: If you want to try the same DX; try importing from  '@aws-amplify/backend/types/platform'

import * as types from '@aws-amplify/backend/types/platform';

Comment (sobolk) [reply packages/backend-auth/package.json:17]: To be clear. The goal is to make it as hidden as possible.

We may say that `<root>/internal/*` is standard way of saying "public but don't use it". However, we haven't completed https://github.com/aws-amplify/amplify-backend/issues/1533 . So either we should define that first or treat this PR as precedent.

**Comment by awsluja** [comment]
Closing this after further review, we have other ways of doing this that don't require custom exports.

**Pull request closed without merging**